projects/batfish-common-protocol/src/main/java/org/batfish/datamodel/pojo/Environment.java

```diff
@@ -0,0 +1,132 @@
+package org.batfish.datamodel.pojo;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import javax.annotation.Nullable;
+import org.batfish.datamodel.Edge;
+import org.batfish.datamodel.collections.NodeInterfacePair;
+
+/**
+ * The {@link Environment Environment} is an Object representation of the environment for BatFish
+ * service.
+ *
+ * <p>Each {@link Environment Environment} contains a name and all informations of the environment
+ * {@link #_name}.
+ */
+public class Environment {
+  private static final String PROP_NAME = "name";
+  private static final String PROP_EDGE_BLACKLIST = "edgeBlacklist";
+  private static final String PROP_INTERFACE_BLACKLIST = "interfaceBlacklist";
+  private static final String PROP_NODE_BLACKLIST = "nodeBlacklist";
+  private static final String PROP_ENVIRONMENT_BGP_TABLES = "bgpTables";
+  private static final String PROP_ENVIRONMENT_ROUTING_TABLES = "routingTables";
+  private static final String PROP_EXTERNAL_BGP_ANNOUNCEMENTS = "externalBgpAnnouncements";
+
+  private final String _name;
+  private final List<Edge> _edgeBlacklist;
+  private final List<NodeInterfacePair> _interfaceBlacklist;
+  private final List<String> _nodeBlacklist;
+  private final Map<String, String> _bgpTables;
+  private final Map<String, String> _routingTables;
+  private final String _externalBgpAnnouncements;
+
+  @JsonCreator
+  public Environment(
+      @JsonProperty(PROP_NAME) String name,
+      @JsonProperty(PROP_EDGE_BLACKLIST) @Nullable List<Edge> edgeBlacklist,
+      @JsonProperty(PROP_INTERFACE_BLACKLIST) @Nullable List<NodeInterfacePair> interfaceBlacklist,
+      @JsonProperty(PROP_NODE_BLACKLIST) @Nullable List<String> nodeBlacklist,
+      @JsonProperty(PROP_ENVIRONMENT_BGP_TABLES) @Nullable Map<String, String> bgpTables,
+      @JsonProperty(PROP_ENVIRONMENT_ROUTING_TABLES) @Nullable Map<String, String> routingTables,
+      @JsonProperty(PROP_EXTERNAL_BGP_ANNOUNCEMENTS) @Nullable String externalBgpAnnouncements) {
+    this._name = name;
+    this._edgeBlacklist = edgeBlacklist == null ? new ArrayList<>() : edgeBlacklist;
```

Comment:
I think so. For these files, we may want to be able to distinguish between file not present and an empty file. The former means blacklist wasn't specified, and the latter means that no (link/node) was blacklisted.